Add icon to exe file in windows export

masoudbh3 opened the pull request:
add version_info and icon sections in "export to windows platform".
add version_info and icon to godot exe file (editor & template exe).
fix an problem in image class.
change all default icons to android export icon (a little more rounded).
create an python script for convert file to cpp byte array for use in
'splash.h'.

export to windows:
![windows-export](https://cloud.githubusercontent.com/assets/9015847/10060689/9d9f9fe8-625f-11e5-8323-4cd597399e76.png)

icon:
![godot_icon](https://cloud.githubusercontent.com/assets/9015847/10060692/a4fb0b56-625f-11e5-93e5-db88d95e71ce.png)

reduz commented:
Wow, amazing! Is this portable? Can I export from linux and it will add the
Icon too?

Also I believe the right way to add the library is in /tools instead of
drivers/
So when you compile without tools it's not included.
On Sep 23, 2015 19:29, "Masoud BH" <notifications@github.com> wrote:

> add version_info and icon sections in "export to windows platform".
> add version_info and icon to godot exe file (editor & template exe).
> fix an problem in image class.
> change all default icons to android export icon (a little more rounded).
> create an python script for convert file to cpp byte array for use in
> 'splash.h'.
>
> export to windows:
> [image: windows-export]
> <https://cloud.githubusercontent.com/assets/9015847/10060689/9d9f9fe8-625f-11e5-8323-4cd597399e76.png>
>
> icon:
> [image: godot_icon]
> <https://cloud.githubusercontent.com/assets/9015847/10060692/a4fb0b56-625f-11e5-93e5-db88d95e71ce.png>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/okamstudio/godot/pull/2518
> Commit Summary
>
>    - Add icon to exe file in windows export
>
> File Changes
>
>    - *M* core/image.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-0> (2)
>    - *M* drivers/SCsub
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-1> (1)
>    - *A* drivers/pe_bliss/README
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-2> (61)
>    - *A* drivers/pe_bliss/SCsub
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-3> (5)
>    - *A* drivers/pe_bliss/entropy.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-4> (90)
>    - *A* drivers/pe_bliss/entropy.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-5> (30)
>    - *A* drivers/pe_bliss/file_version_info.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-6> (419)
>    - *A* drivers/pe_bliss/file_version_info.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-7> (178)
>    - *A* drivers/pe_bliss/message_table.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-8> (60)
>    - *A* drivers/pe_bliss/message_table.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-9> (35)
>    - *A* drivers/pe_bliss/pe_base.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-10> (1659)
>    - *A* drivers/pe_bliss/pe_base.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-11> (523)
>    - *A* drivers/pe_bliss/pe_bliss.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-12> (18)
>    - *A* drivers/pe_bliss/pe_bliss_godot.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-13> (118)
>    - *A* drivers/pe_bliss/pe_bliss_godot.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-14> (7)
>    - *A* drivers/pe_bliss/pe_bliss_resources.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-15> (15)
>    - *A* drivers/pe_bliss/pe_bound_import.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-16> (290)
>    - *A* drivers/pe_bliss/pe_bound_import.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-17> (87)
>    - *A* drivers/pe_bliss/pe_checksum.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-18> (82)
>    - *A* drivers/pe_bliss/pe_checksum.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-19> (9)
>    - *A* drivers/pe_bliss/pe_debug.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-20> (844)
>    - *A* drivers/pe_bliss/pe_debug.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-21> (303)
>    - *A* drivers/pe_bliss/pe_directory.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-22> (38)
>    - *A* drivers/pe_bliss/pe_directory.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-23> (29)
>    - *A* drivers/pe_bliss/pe_dotnet.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-24> (165)
>    - *A* drivers/pe_bliss/pe_dotnet.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-25> (76)
>    - *A* drivers/pe_bliss/pe_exception.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-26> (19)
>    - *A* drivers/pe_bliss/pe_exception.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-27> (109)
>    - *A* drivers/pe_bliss/pe_exception_directory.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-28> (156)
>    - *A* drivers/pe_bliss/pe_exception_directory.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-29> (67)
>    - *A* drivers/pe_bliss/pe_exports.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-30> (679)
>    - *A* drivers/pe_bliss/pe_exports.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-31> (163)
>    - *A* drivers/pe_bliss/pe_factory.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-32> (22)
>    - *A* drivers/pe_bliss/pe_factory.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-33> (18)
>    - *A* drivers/pe_bliss/pe_imports.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-34> (756)
>    - *A* drivers/pe_bliss/pe_imports.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-35> (187)
>    - *A* drivers/pe_bliss/pe_load_config.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-36> (536)
>    - *A* drivers/pe_bliss/pe_load_config.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-37> (163)
>    - *A* drivers/pe_bliss/pe_properties.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-38> (20)
>    - *A* drivers/pe_bliss/pe_properties.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-39> (215)
>    - *A* drivers/pe_bliss/pe_properties_generic.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-40> (624)
>    - *A* drivers/pe_bliss/pe_properties_generic.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-41> (256)
>    - *A* drivers/pe_bliss/pe_rebuilder.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-42> (193)
>    - *A* drivers/pe_bliss/pe_rebuilder.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-43> (19)
>    - *A* drivers/pe_bliss/pe_relocations.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-44> (299)
>    - *A* drivers/pe_bliss/pe_relocations.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-45> (101)
>    - *A* drivers/pe_bliss/pe_resource_manager.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-46> (265)
>    - *A* drivers/pe_bliss/pe_resource_manager.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-47> (92)
>    - *A* drivers/pe_bliss/pe_resource_viewer.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-48> (361)
>    - *A* drivers/pe_bliss/pe_resource_viewer.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-49> (132)
>    - *A* drivers/pe_bliss/pe_resources.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-50> (705)
>    - *A* drivers/pe_bliss/pe_resources.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-51> (224)
>    - *A* drivers/pe_bliss/pe_rich_data.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-52> (131)
>    - *A* drivers/pe_bliss/pe_rich_data.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-53> (37)
>    - *A* drivers/pe_bliss/pe_section.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-54> (281)
>    - *A* drivers/pe_bliss/pe_section.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-55> (137)
>    - *A* drivers/pe_bliss/pe_structures.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-56> (1007)
>    - *A* drivers/pe_bliss/pe_tls.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-57> (375)
>    - *A* drivers/pe_bliss/pe_tls.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-58> (101)
>    - *A* drivers/pe_bliss/resource_bitmap_reader.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-59> (65)
>    - *A* drivers/pe_bliss/resource_bitmap_reader.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-60> (29)
>    - *A* drivers/pe_bliss/resource_bitmap_writer.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-61> (54)
>    - *A* drivers/pe_bliss/resource_bitmap_writer.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-62> (26)
>    - *A* drivers/pe_bliss/resource_cursor_icon_reader.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-63> (500)
>    - *A* drivers/pe_bliss/resource_cursor_icon_reader.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-64> (63)
>    - *A* drivers/pe_bliss/resource_cursor_icon_writer.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-65> (426)
>    - *A* drivers/pe_bliss/resource_cursor_icon_writer.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-66> (73)
>    - *A* drivers/pe_bliss/resource_data_info.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-67> (27)
>    - *A* drivers/pe_bliss/resource_data_info.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-68> (27)
>    - *A* drivers/pe_bliss/resource_internal.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-69> (13)
>    - *A* drivers/pe_bliss/resource_message_list_reader.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-70> (110)
>    - *A* drivers/pe_bliss/resource_message_list_reader.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-71> (28)
>    - *A* drivers/pe_bliss/resource_string_table_reader.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-72> (88)
>    - *A* drivers/pe_bliss/resource_string_table_reader.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-73> (36)
>    - *A* drivers/pe_bliss/resource_version_info_reader.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-74> (290)
>    - *A* drivers/pe_bliss/resource_version_info_reader.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-75> (46)
>    - *A* drivers/pe_bliss/resource_version_info_writer.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-76> (262)
>    - *A* drivers/pe_bliss/resource_version_info_writer.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-77> (31)
>    - *A* drivers/pe_bliss/stdint_defs.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-78> (24)
>    - *A* drivers/pe_bliss/utils.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-79> (86)
>    - *A* drivers/pe_bliss/utils.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-80> (84)
>    - *A* drivers/pe_bliss/version_info_editor.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-81> (163)
>    - *A* drivers/pe_bliss/version_info_editor.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-82> (58)
>    - *A* drivers/pe_bliss/version_info_types.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-83> (17)
>    - *A* drivers/pe_bliss/version_info_viewer.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-84> (159)
>    - *A* drivers/pe_bliss/version_info_viewer.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-85> (68)
>    - *M* drivers/register_driver_types.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-86> (11)
>    - *A* godot_icon.png
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-87> (0)
>    - *A* godot_icon.svg
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-88> (133)
>    - *M* main/splash.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-89> (2)
>    - *M* methods.py
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-90> (2)
>    - *M* platform/android/java/res/drawable/icon.png
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-91> (0)
>    - *M* platform/nacl/html/icon_128.png
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-92> (0)
>    - *M* platform/nacl/html/icon_16.png
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-93> (0)
>    - *M* platform/windows/SCsub
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-94> (8)
>    - *M* platform/windows/detect.py
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-95> (31)
>    - *M* platform/windows/export/export.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-96> (349)
>    - *M* platform/windows/export/export.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-97> (35)
>    - *A* platform/windows/godot.ico
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-98> (0)
>    - *A* platform/windows/godot_res.rc
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-99> (33)
>    - *M* tools/Godot.app/Contents/Resources/Godot.icns
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-100> (0)
>    - *M* tools/osx_template.app/Contents/Resources/icon.icns
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-101> (0)
>    - *A* tools/scripts/file-hex-array.py
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-102> (52)
>
> Patch Links:
>
>    - https://github.com/okamstudio/godot/pull/2518.patch
>    - https://github.com/okamstudio/godot/pull/2518.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/pull/2518>.
>

okamstudio commented:
Oh, splash can already be customized in windows
On Sep 23, 2015 19:29, "Masoud BH" <notifications@github.com> wrote:

> add version_info and icon sections in "export to windows platform".
> add version_info and icon to godot exe file (editor & template exe).
> fix an problem in image class.
> change all default icons to android export icon (a little more rounded).
> create an python script for convert file to cpp byte array for use in
> 'splash.h'.
>
> export to windows:
> [image: windows-export]
> <https://cloud.githubusercontent.com/assets/9015847/10060689/9d9f9fe8-625f-11e5-8323-4cd597399e76.png>
>
> icon:
> [image: godot_icon]
> <https://cloud.githubusercontent.com/assets/9015847/10060692/a4fb0b56-625f-11e5-93e5-db88d95e71ce.png>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/okamstudio/godot/pull/2518
> Commit Summary
>
>    - Add icon to exe file in windows export
>
> File Changes
>
>    - *M* core/image.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-0> (2)
>    - *M* drivers/SCsub
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-1> (1)
>    - *A* drivers/pe_bliss/README
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-2> (61)
>    - *A* drivers/pe_bliss/SCsub
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-3> (5)
>    - *A* drivers/pe_bliss/entropy.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-4> (90)
>    - *A* drivers/pe_bliss/entropy.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-5> (30)
>    - *A* drivers/pe_bliss/file_version_info.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-6> (419)
>    - *A* drivers/pe_bliss/file_version_info.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-7> (178)
>    - *A* drivers/pe_bliss/message_table.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-8> (60)
>    - *A* drivers/pe_bliss/message_table.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-9> (35)
>    - *A* drivers/pe_bliss/pe_base.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-10> (1659)
>    - *A* drivers/pe_bliss/pe_base.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-11> (523)
>    - *A* drivers/pe_bliss/pe_bliss.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-12> (18)
>    - *A* drivers/pe_bliss/pe_bliss_godot.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-13> (118)
>    - *A* drivers/pe_bliss/pe_bliss_godot.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-14> (7)
>    - *A* drivers/pe_bliss/pe_bliss_resources.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-15> (15)
>    - *A* drivers/pe_bliss/pe_bound_import.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-16> (290)
>    - *A* drivers/pe_bliss/pe_bound_import.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-17> (87)
>    - *A* drivers/pe_bliss/pe_checksum.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-18> (82)
>    - *A* drivers/pe_bliss/pe_checksum.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-19> (9)
>    - *A* drivers/pe_bliss/pe_debug.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-20> (844)
>    - *A* drivers/pe_bliss/pe_debug.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-21> (303)
>    - *A* drivers/pe_bliss/pe_directory.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-22> (38)
>    - *A* drivers/pe_bliss/pe_directory.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-23> (29)
>    - *A* drivers/pe_bliss/pe_dotnet.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-24> (165)
>    - *A* drivers/pe_bliss/pe_dotnet.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-25> (76)
>    - *A* drivers/pe_bliss/pe_exception.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-26> (19)
>    - *A* drivers/pe_bliss/pe_exception.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-27> (109)
>    - *A* drivers/pe_bliss/pe_exception_directory.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-28> (156)
>    - *A* drivers/pe_bliss/pe_exception_directory.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-29> (67)
>    - *A* drivers/pe_bliss/pe_exports.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-30> (679)
>    - *A* drivers/pe_bliss/pe_exports.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-31> (163)
>    - *A* drivers/pe_bliss/pe_factory.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-32> (22)
>    - *A* drivers/pe_bliss/pe_factory.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-33> (18)
>    - *A* drivers/pe_bliss/pe_imports.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-34> (756)
>    - *A* drivers/pe_bliss/pe_imports.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-35> (187)
>    - *A* drivers/pe_bliss/pe_load_config.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-36> (536)
>    - *A* drivers/pe_bliss/pe_load_config.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-37> (163)
>    - *A* drivers/pe_bliss/pe_properties.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-38> (20)
>    - *A* drivers/pe_bliss/pe_properties.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-39> (215)
>    - *A* drivers/pe_bliss/pe_properties_generic.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-40> (624)
>    - *A* drivers/pe_bliss/pe_properties_generic.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-41> (256)
>    - *A* drivers/pe_bliss/pe_rebuilder.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-42> (193)
>    - *A* drivers/pe_bliss/pe_rebuilder.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-43> (19)
>    - *A* drivers/pe_bliss/pe_relocations.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-44> (299)
>    - *A* drivers/pe_bliss/pe_relocations.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-45> (101)
>    - *A* drivers/pe_bliss/pe_resource_manager.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-46> (265)
>    - *A* drivers/pe_bliss/pe_resource_manager.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-47> (92)
>    - *A* drivers/pe_bliss/pe_resource_viewer.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-48> (361)
>    - *A* drivers/pe_bliss/pe_resource_viewer.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-49> (132)
>    - *A* drivers/pe_bliss/pe_resources.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-50> (705)
>    - *A* drivers/pe_bliss/pe_resources.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-51> (224)
>    - *A* drivers/pe_bliss/pe_rich_data.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-52> (131)
>    - *A* drivers/pe_bliss/pe_rich_data.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-53> (37)
>    - *A* drivers/pe_bliss/pe_section.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-54> (281)
>    - *A* drivers/pe_bliss/pe_section.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-55> (137)
>    - *A* drivers/pe_bliss/pe_structures.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-56> (1007)
>    - *A* drivers/pe_bliss/pe_tls.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-57> (375)
>    - *A* drivers/pe_bliss/pe_tls.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-58> (101)
>    - *A* drivers/pe_bliss/resource_bitmap_reader.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-59> (65)
>    - *A* drivers/pe_bliss/resource_bitmap_reader.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-60> (29)
>    - *A* drivers/pe_bliss/resource_bitmap_writer.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-61> (54)
>    - *A* drivers/pe_bliss/resource_bitmap_writer.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-62> (26)
>    - *A* drivers/pe_bliss/resource_cursor_icon_reader.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-63> (500)
>    - *A* drivers/pe_bliss/resource_cursor_icon_reader.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-64> (63)
>    - *A* drivers/pe_bliss/resource_cursor_icon_writer.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-65> (426)
>    - *A* drivers/pe_bliss/resource_cursor_icon_writer.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-66> (73)
>    - *A* drivers/pe_bliss/resource_data_info.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-67> (27)
>    - *A* drivers/pe_bliss/resource_data_info.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-68> (27)
>    - *A* drivers/pe_bliss/resource_internal.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-69> (13)
>    - *A* drivers/pe_bliss/resource_message_list_reader.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-70> (110)
>    - *A* drivers/pe_bliss/resource_message_list_reader.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-71> (28)
>    - *A* drivers/pe_bliss/resource_string_table_reader.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-72> (88)
>    - *A* drivers/pe_bliss/resource_string_table_reader.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-73> (36)
>    - *A* drivers/pe_bliss/resource_version_info_reader.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-74> (290)
>    - *A* drivers/pe_bliss/resource_version_info_reader.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-75> (46)
>    - *A* drivers/pe_bliss/resource_version_info_writer.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-76> (262)
>    - *A* drivers/pe_bliss/resource_version_info_writer.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-77> (31)
>    - *A* drivers/pe_bliss/stdint_defs.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-78> (24)
>    - *A* drivers/pe_bliss/utils.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-79> (86)
>    - *A* drivers/pe_bliss/utils.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-80> (84)
>    - *A* drivers/pe_bliss/version_info_editor.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-81> (163)
>    - *A* drivers/pe_bliss/version_info_editor.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-82> (58)
>    - *A* drivers/pe_bliss/version_info_types.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-83> (17)
>    - *A* drivers/pe_bliss/version_info_viewer.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-84> (159)
>    - *A* drivers/pe_bliss/version_info_viewer.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-85> (68)
>    - *M* drivers/register_driver_types.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-86> (11)
>    - *A* godot_icon.png
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-87> (0)
>    - *A* godot_icon.svg
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-88> (133)
>    - *M* main/splash.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-89> (2)
>    - *M* methods.py
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-90> (2)
>    - *M* platform/android/java/res/drawable/icon.png
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-91> (0)
>    - *M* platform/nacl/html/icon_128.png
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-92> (0)
>    - *M* platform/nacl/html/icon_16.png
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-93> (0)
>    - *M* platform/windows/SCsub
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-94> (8)
>    - *M* platform/windows/detect.py
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-95> (31)
>    - *M* platform/windows/export/export.cpp
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-96> (349)
>    - *M* platform/windows/export/export.h
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-97> (35)
>    - *A* platform/windows/godot.ico
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-98> (0)
>    - *A* platform/windows/godot_res.rc
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-99> (33)
>    - *M* tools/Godot.app/Contents/Resources/Godot.icns
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-100> (0)
>    - *M* tools/osx_template.app/Contents/Resources/icon.icns
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-101> (0)
>    - *A* tools/scripts/file-hex-array.py
>    <https://github.com/okamstudio/godot/pull/2518/files#diff-102> (52)
>
> Patch Links:
>
>    - https://github.com/okamstudio/godot/pull/2518.patch
>    - https://github.com/okamstudio/godot/pull/2518.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/pull/2518>.
>

masoudbh3 commented:
yes i use pe_bliss library for do that (windows and linux).
and for compile godot_res.rc to res file , i write some codes for support mingw.

masoudbh3 commented:
but I thought all libraries there In /drivers .
i add this code in drivers SCsub :
```if (env["tools"]=="yes"):
	SConscript('pe_bliss/SCsub');```

reduz commented:
Yeah it's probably not very clear, for reference freetype (which is used
for the font importer) is in tools/freetype, so I guess simply move it over
to tools/pe_bliss

On Wed, Sep 23, 2015 at 7:51 PM, Masoud BH <notifications@github.com> wrote:

> but I thought all libraries there In /drivers .
> i add this code in drivers SCsub :
> if (env["tools"]=="yes"):
> SConscript('pe_bliss/SCsub');
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/pull/2518#issuecomment-142753607>.
>

masoudbh3 committed:
change pe_bliss parent directory from /drivers to /tools

punto- commented:
Do we know the authors of this "pe_bliss" library? can we get them to
give us the library with a more standard license, like MIT, BSD, etc?
the statement "Free to use, modify, and distribute." on their readme
might not be enough.

On 24 September 2015 at 00:53, Juan Linietsky <notifications@github.com> wrote:
> Yeah it's probably not very clear, for reference freetype (which is used
> for the font importer) is in tools/freetype, so I guess simply move it over
> to tools/pe_bliss
>
> On Wed, Sep 23, 2015 at 7:51 PM, Masoud BH <notifications@github.com> wrote:
>
>> but I thought all libraries there In /drivers .
>> i add this code in drivers SCsub :
>> if (env["tools"]=="yes"):
>> SConscript('pe_bliss/SCsub');
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/okamstudio/godot/pull/2518#issuecomment-142753607>.
>
>>
>
> —
> Reply to this email directly or view it on GitHub.

masoudbh3 commented:
i found pe_bliss library from http://code.google.com/p/portable-executable-library

akien-mga commented:
>  change all default icons to android export icon (a little more rounded).

Will it be possible to disable this? Personally when I see a desktop game with a rounded icon I think "damn yet another mobile game ported to desktop, not interested". Please give us the possibility to keep our good old square icons on desktop :D

punto- commented:
@masoudbh3 where does the README file come from? I don't see it on the google code repository

mrezai commented:
From https://code.google.com/p/portable-executable-library/ :
"Library is free to use in both commertial and non-commertial projects. You can also modify and redistribute it. If you are using it, please, do not forget to specify the name or other copyright of PE Bliss somewhere in the description of your project."
There is a blog post in russian about the project here : http://kaimi.ru/2012/09/portable-executable-library/

akien-mga commented:
>  "Library is free to use in both commertial and non-commertial projects. You can also modify and redistribute it. If you are using it, please, do not forget to specify the name or other copyright of PE Bliss somewhere in the description of your project."

The rights granted seem to be good, but it's still a custom license, and mentioned only on a Google Code page that is already archived (and who knows how long Google will keep the archives up), so I would advise against using this. We probably should avoid "tainted" Godot's MIT code with external libraries using non-standard licenses.

reduz commented:
Can't the author be contacted?


On Fri, Sep 25, 2015 at 12:23 PM, Rémi Verschelde <notifications@github.com>
wrote:

> "Library is free to use in both commertial and non-commertial projects.
> You can also modify and redistribute it. If you are using it, please, do
> not forget to specify the name or other copyright of PE Bliss somewhere in
> the description of your project."
>
> The rights granted seem to be good, but it's still a custom license, and
> mentioned only on a Google Code page that is already archived (and who
> knows how long Google will keep the archives up), so I would advise against
> using this. We probably should avoid "tainted" Godot's MIT code with
> external libraries using non-standard licenses.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/pull/2518#issuecomment-143251072>.
>

reduz commented:
Google says his mail is: rukaimi@gmail.com
We could ask him to relicense it under MIT


On Fri, Sep 25, 2015 at 12:51 PM, Juan Linietsky <reduzio@gmail.com> wrote:

> Can't the author be contacted?
>
>
> On Fri, Sep 25, 2015 at 12:23 PM, Rémi Verschelde <
> notifications@github.com> wrote:
>
>> "Library is free to use in both commertial and non-commertial projects.
>> You can also modify and redistribute it. If you are using it, please, do
>> not forget to specify the name or other copyright of PE Bliss somewhere in
>> the description of your project."
>>
>> The rights granted seem to be good, but it's still a custom license, and
>> mentioned only on a Google Code page that is already archived (and who
>> knows how long Google will keep the archives up), so I would advise against
>> using this. We probably should avoid "tainted" Godot's MIT code with
>> external libraries using non-standard licenses.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/okamstudio/godot/pull/2518#issuecomment-143251072>.
>>
>
>

akien-mga commented:
> Can't the author be contacted?
> We could ask him to relicense it under MIT

That's always a good idea IMO :)

punto- commented:
I wrote to him last night..

On 25 September 2015 at 17:52, Juan Linietsky <notifications@github.com>
wrote:

> Google says his mail is: rukaimi@gmail.com
> We could ask him to relicense it under MIT
>
>
> On Fri, Sep 25, 2015 at 12:51 PM, Juan Linietsky <reduzio@gmail.com>
> wrote:
>
> > Can't the author be contacted?
> >
> >
> > On Fri, Sep 25, 2015 at 12:23 PM, Rémi Verschelde <
> > notifications@github.com> wrote:
> >
> >> "Library is free to use in both commertial and non-commertial projects.
> >> You can also modify and redistribute it. If you are using it, please, do
> >> not forget to specify the name or other copyright of PE Bliss somewhere
> in
> >> the description of your project."
> >>
> >> The rights granted seem to be good, but it's still a custom license, and
> >> mentioned only on a Google Code page that is already archived (and who
> >> knows how long Google will keep the archives up), so I would advise
> against
> >> using this. We probably should avoid "tainted" Godot's MIT code with
> >> external libraries using non-standard licenses.
> >>
> >> —
> >> Reply to this email directly or view it on GitHub
> >> <https://github.com/okamstudio/godot/pull/2518#issuecomment-143251072>.
>
> >>
> >
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/pull/2518#issuecomment-143259436>.
>

masoudbh3 commented:
@punto- there is an russian readme.txt file in /pe_lib direcotory.
@akien-mga no problem you can change your game icon when export that.
@reduz i wrote to kaimi when post this commit, but not given answer.

other contact info for kaimi:
ICQ: 1003875
Jabber: kaimi@kdetalk.net

from readme file i found author is DX not kaimi.
he/she is another kaimi.ru blog author.
but does not exist any contact info except send PM to this :
http://forum.kaimi.ru/profile.php?id=3
https://translate.google.com/translate?hl=en&sl=ru&tl=en&u=https%3A%2F%2Fforum.kaimi.ru%2Fprofile.php%3Fid%3D3&sandbox=1

punto- commented:
I think there's 2 questions here:
1) does the license text apply to the whole library? I think yes, that
is obviously the intention, but some more strict people might
interpret it as no, unless it's stated on every source file or a
COPYING or LICENSE file or something like that (but I don't know what
exactly). It also doesn't help that we find this "licence-y" text in 3
different places (the README, the header pe_base.h and the Google Code
page), and on each the text is slightly different.
2) if we assume that 1) is good, the second question is, is the
license compatible with MIT? I think yes, the intention seems to be
there, but given my experience[1] I agree with Rémi that it might not
be good enough.

This sucks because we really appreciate this work, and we've been
trying to get these features for a while, but ideally we'd have to get
the author of the library to give us a standard license for it.

[1] I used to maintain the tolua++ library, which was a fork of tolua,
which had a similar custom license, and we used to get requests all
the time from medium and big game dev studios to give them a standard
license or a custom license (ie a signed contract by me saying I won't
sue them :p ). Eventually we switched to MIT with permission from the
original author. The game development industry is kinda iffy about
custom licenses, but they're perfectly fine with MIT, etc.


On 25 September 2015 at 17:23, Rémi Verschelde <notifications@github.com> wrote:
> "Library is free to use in both commertial and non-commertial projects. You
> can also modify and redistribute it. If you are using it, please, do not
> forget to specify the name or other copyright of PE Bliss somewhere in the
> description of your project."
>
> The rights granted seem to be good, but it's still a custom license, and
> mentioned only on a Google Code page that is already archived (and who knows
> how long Google will keep the archives up), so I would advise against using
> this. We probably should avoid "tainted" Godot's MIT code with external
> libraries using non-standard licenses.
>
> —
> Reply to this email directly or view it on GitHub.


On 25 September 2015 at 12:23, Rémi Verschelde <notifications@github.com> wrote:
> "Library is free to use in both commertial and non-commertial projects. You
> can also modify and redistribute it. If you are using it, please, do not
> forget to specify the name or other copyright of PE Bliss somewhere in the
> description of your project."
>
> The rights granted seem to be good, but it's still a custom license, and
> mentioned only on a Google Code page that is already archived (and who knows
> how long Google will keep the archives up), so I would advise against using
> this. We probably should avoid "tainted" Godot's MIT code with external
> libraries using non-standard licenses.
>
> —
> Reply to this email directly or view it on GitHub.

reduz commented:
Do we have any news about the state of this?

sezit commented:
Here reply. But who asked i dont know.
![default](https://cloud.githubusercontent.com/assets/6241034/10561414/539867fc-7530-11e5-8771-1096fc250777.PNG)

masoudbh3 commented:
Who has the answer?

punto- commented:
I got permission from the author to add the MIT license text to all the
files in the library, so let's do that on this PR for the copy that we
would integrate on our repo. This is the text:

--
Copyright (c) 2015 dx, http://kaimi.ru

Permission is hereby granted, free of charge, to any person obtaining
a copy of this software and associated documentation files (the
"Software"), to deal in the Software without restriction, including
without limitation the rights to use, copy, modify, merge, publish,
distribute, sublicense, and/or sell copies of the Software, and to
permit persons to whom the Software is furnished to do so, subject to
the following conditions:

The above copyright notice and this permission notice shall be
included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
--

This needs to go on all the files that belong to the pe_bliss library.

Then we wait to hear from the legal people to see if this is acceptable.



On 19 October 2015 at 17:25, Masoud BH <notifications@github.com> wrote:

> Who has the answer?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/pull/2518#issuecomment-149335957>.
>

masoudbh3 committed:
Add icon to exe file in windows export

add version_info and icon sections in "export to windows platform".
add version_info and icon to godot exe file (editor & template exe).
fix an problem in image class.
change all default icons to android export icon (a little more rounded).
create an python script for convert file to cpp byte array for use in
'splash.h'.

masoudbh3 committed:
add copyright text

reduz commented:
OK since we got permission from the author of pe_bliss, is this ready to be merged?

masoudbh3 commented:
@punto- got permission and i add license text to library files.
maybe ready for merge now. i don't know.
